### ISSUE TITLE
molecule test --all should destroy all scenarios before starting the full matrix

### ISSUE THREAD

**Issue by blackillzone** [metacloud/molecule#1076]
# Issue Type

- Bug report

# Molecule and Ansible details

```
ansible --version
ansible 2.4.2.0
---
molecule --version
molecule, version 2.5.0
```

- Molecule installation method: pip
- Ansible installation method: pip

# Desired Behaviour

`molecule test --all` should do a `molecule destroy --all` before.

# Actual Behaviour (Bug report only)

Currently, doing a `molecule test --all` start all the scenarios, following a matrix. If, for example, the molecule command failed during the second scenario, relaunching the `molecule test --all` will not destroy the VMs or containers of this scenario. This will potentially give incoherences in the first scenario, or even conflicts.

I don't know if it's possible to add this into molecule test command. I know I can do a `molecule destroy -all` before each of my `molecule test --all`, but as the `molecule test` already do a destroy, I think it's overkill.

What do you think of this ?

I just though that I also could alter the test matrix to remove the destroy step, and launch a `destroy --all` before. I don't know what is better though.

**Comment by retr0h** [comment]
`molecule test --all` will execute all scenarios.  However, it does not destroy the scenarios all up front.  Molecule loops through each scenario performing the steps in defined in `test_sequence`.  The test sequence contains the following.

```
                'test_sequence': [
                    'lint',
                    'destroy',
                    'dependency',
                    'syntax',
                    'create',
                    'prepare',
                    'converge',
                    'idempotence',
                    'side_effect',
                    'verify',
                    'destroy',
                ],
```

Basically the `--all` flag just tells Molecule to execute all scenarios vs the default scenario `default`.

**Comment by blackillzone** [comment]
Ok I see, and the last command do the trick for me. Thank you ! But don't you think this should be as a default option in `molecule test --all`, instead of never ?
It seems more user-friendly for me to clean everything by default, when you launch a full test, instead of adding the `--destroy always`.

**Comment by retr0h** [comment]
When something goes wrong, I generally want to login and see why.

**Comment by retr0h** [comment]
You are right.  That was a bug that I introduced.  Fixed in #1086.  My apologies for not fully understanding.
